robot/client/client.go

```diff
@@ -1078,3 +1083,54 @@ func (rc *RobotClient) Version(ctx context.Context) (robot.VersionResponse, erro
 
 	return mVersion, nil
 }
+
+func addVersionMetadataToContext(ctx context.Context) context.Context {
+	info, _ := debug.ReadBuildInfo()
+	deps := make(map[string]*debug.Module, len(info.Deps))
+	for _, dep := range info.Deps {
+		deps[dep.Path] = dep
+	}
+	var apiVersion string
+	if dep, ok := deps["go.viam.com/api"]; ok {
```

Comment:
Out of curiosity, have you tested all of this and confirmed the output? I don't know exactly how `ReadBuildInfo` works under the hood but it looks to me like `go.viam.com/api` points to our github repo, which has a version tag that's pretty out of date relative to what's in `buf.build`. But, I could be misunderstanding where the `Deps` info comes from!

Reply:
I did confirm the output, and it is behind. `go.viam.com/api` does point to our github repo, but this is another thing I do not appreciate about having the Go SDK and RDK together. The RDK does nothing with the `protos-updated` github workflow from what I can see. All other mentions of the API use this method, so either they're all doing it wrong or the Go SDK does not update like the others.

Reply:
It's a little difficult to understand the flow of the Go client because of how mixed up everything is with each other 😓

Reply:
`v0.1.330` isn't a released tag, so I guess it is grabbing from something else.

Reply:
is `v0.1.330` the version it gave you? because if so that's ideal!